map.go

```diff
@@ -717,13 +754,34 @@ func (mi *MapIterator) Next(keyOut, valueOut interface{}) bool {
 		return false
 	}
 
+	// gessing a random start key
+	if !supportNullKey.value && mi.randomStartGuess {
+		mi.randomStartGuess = false
+
+		var bytes1, bytes2, bytes3 []byte
+		bytes1, mi.err = mi.target.NextKeyBytes(mi.target.randKey)
+		if mi.err != nil {
+			return false
+		}
+		mi.prevKey = mi.target.randomizeKey()
+		bytes2, mi.err = mi.target.NextKeyBytes(mi.prevKey)
+		if !bytes.Equal(bytes1, bytes2) {
+			mi.prevKey = mi.target.randomizeKey()
+			bytes3, mi.err = mi.target.NextKeyBytes(mi.prevKey)
+			if !bytes.Equal(bytes1, bytes3) && !bytes.Equal(bytes2, bytes3) {
+				// we didn't found a random key
+				mi.err = fmt.Errorf("%w", ErrIterationAborted)
```

Comment:
I don't think you need to wrap the error here. Couldn't you do `mi.err = ErrIterationAborted`?

Reply:
They use wrapped error logic here, so errors.Is() would work
that's the same return way than line 818 (same function)

Reply:
I don't get the error.wrap logic of the ebpf package here
 fmt.Errorf("%w",...) would make the error not exposed to outside
 but they use ErrIter... and expose the variable/error outside of the package

I agree with you we should return `ErrIterationAborted` directly
errors.Is() would work anyway as errors.New() (errorString{}) is Comparable()

Same remark/behaviour on syscalls.go:328

To conclude and may a go vet should exist for this, a fmt.Errorf("%w")   +  ExportedVar as argument make no such sense

Reply:
I would leave the code as is for this. As it should be addressed to cilium/ebpf and as we plan to move to the new dd/ebpf-manager pkg

Reply:
They do it because they don't want folks writing `if err == ErrIterationAborted`, but `if errors.Is(err, ErrIterationAborted)`. Returning `fmt.Errorf("%w", ErrIterationAborted)` forces the second case.